release-content/0.16/release-notes/gpu_driven_rendering.md

```diff
@@ -0,0 +1,75 @@
+
+To explain how GPU-driven rendering operates, it's easiest to first describe how CPU-driven rendering works:
+
+1. The CPU determines the objects that are visible, via frustum culling and perhaps occlusion culling.
+
+2. For each such object:
+
+   a. The CPU sends the object's transform to the GPU, possibly in addition to other data such as joint weights.
+
+   b. The CPU tells the GPU where the mesh data is.
+
+   c. The CPU writes the material data to the GPU.
+
+   d. The CPU tells the GPU where textures and other buffers needed to render the objects are (light data, etc.)
+
+   e. The CPU issues a drawcall.
+
+   f. The GPU renders the object.
+
+In contrast, GPU-driven rendering in Bevy works like this:
+
+1. The CPU uploads transform information for all objects at once to the GPU.
+
+2. If new objects have been spawned since the last frame, the CPU fills out tables specifying where the mesh data for the new objects are.
+
+3. If materials have been modified since the last frame, the CPU uploads information about those materials to the GPU.
+
+4. The CPU creates lists of objects to be rendered this frame. Each object is simply referenced by an integer ID, so these lists are small. The number of lists depends on the size and complexity of the scene, but there are rarely more than 15 such lists even for large scenes.
+
+5. For each such list:
+
+    a. The CPU issues a *single* drawcall.
+
+    b. The GPU processes all objects in the list, determining which ones are truly visible.
+
+    c. The GPU renders each such visible object.
+
+For large scenes that may have tens of thousands of objects, GPU-driven rendering frequently results in reduction in CPU rendering overhead of 3× or more. It's also necessary for occlusion culling, because of the GPU transform step (5(b) above).
+
+Internally, GPU-driven rendering is less a single technique than a collection of techniques. These include:
+
+* *Multi-draw indirect* (MDI), a GPU API that allows multiple meshes to be drawn in a single drawcall, the details of which the GPU provides by filling out tables in GPU memory. In order to use MDI effectively, Bevy uses a new subsystem, the *mesh allocator*, which manages the details of packing meshes together in GPU memory.
+
+  - *Multi-draw indirect count* (MDIC), an extension to multi-draw indirect that allows the GPU to determine the *number* of meshes to draw with minimal overhead.
+
+* *Bindless resources*, which allow Bevy to supply the textures (and other resources) for many objects as a group, instead of having to bind textures one-by-one on the CPU. These resources are managed by a new subsystem known as the *material allocator*.
+
+* *GPU transform and cull*, which allows Bevy to compute the position and visibility of every object from the camera's perspective on the GPU instead of on the GPU.
```

Comment:
```suggestion
* *GPU transform and cull*, which allows Bevy to compute the position and visibility of every object from the camera's perspective on the GPU instead of on the CPU.
```